stdlib/string.gr

```diff
@@ -1,17 +1,31 @@
+/**
+ * @module String: Utilities for working with strings.
+ * @example import String from "string"
+ */
 import WasmI32 from "runtime/unsafe/wasmi32"
 import Memory from "runtime/unsafe/memory"
 import Exception from "runtime/exception"
 import { tagSimpleNumber, allocateArray, allocateChar, allocateString, allocateBytes } from "runtime/dataStructures"
 
-// Concatenate two strings.
-// @param str1: String - The first string
-// @param str2: String - The second string
-// @returns String
+/**
+ * Concatenate two strings.
+ * 
+ * @param str1: The first string
+ * @param str2: The second string
+ * @returns The result of the joined strings
+ * 
+ * @example String.concat("Hello", " world")
+ */
 export let concat = (++)
 
-// Get the character length of a string.
-// @param input: String - The string to check
-// @returns Number
+/**
+ * Get the character length of a string.
+ * 
+ * @param input: The string to check
+ * @returns The length of the provided string
+ * 
+ * @example String.length("Hello world")
+ */
 @disableGC
 export let rec length = (s: String) => {
```

Comment:
```
export let rec length = (string: String) => {
```
You'll need to go through this file and update the arguments to functions to match the names in the docs because it does a lookup between the two.